pkg/actors/internal/placement.go

```diff
@@ -129,70 +146,63 @@ func (p *ActorPlacement) AddHostedActorType(actorType string) error {
 
 // Start connects placement service to register to membership and send heartbeat
 // to report the current member status periodically.
-func (p *ActorPlacement) Start() {
-	p.serverIndex.Store(0)
-	p.shutdown.Store(false)
-
-	if established := p.establishStreamConn(); !established {
-		return
+func (p *ActorPlacement) Start(ctx context.Context) error {
+	if p.shutdown.Load() {
+		return errors.New("placement is shutdown")
 	}
 
-	// establish connection loop, whenever a disconnect occurs it starts to run trying to connect to a new server.
-	p.shutdownConnLoop.Add(1)
-	go func() {
-		defer p.shutdownConnLoop.Done()
-		for !p.shutdown.Load() {
-			// wait until disconnection occurs or shutdown is triggered
-			p.client.waitUntil(func(streamConnAlive bool) bool {
-				return !streamConnAlive || p.shutdown.Load()
-			})
-
-			if p.shutdown.Load() {
-				break
-			}
-			p.establishStreamConn()
+	for {
+		if err := p.manageConnectionLoop(ctx); err != nil {
+			return err
 		}
-	}()
-
-	// Establish receive channel to retrieve placement table update
-	p.shutdownConnLoop.Add(1)
-	go func() {
-		defer p.shutdownConnLoop.Done()
-		for !p.shutdown.Load() {
-			// Wait until stream is connected or shutdown is triggered.
-			p.client.waitUntil(func(streamAlive bool) bool {
-				return streamAlive || p.shutdown.Load()
-			})
-
-			resp, err := p.client.recv()
-			if p.shutdown.Load() {
-				break
-			}
 
-			// TODO: we may need to handle specific errors later.
-			if err != nil {
-				p.client.disconnectFn(func() {
-					p.onPlacementError(err) // depending on the returned error a new server could be used
-				})
-			} else {
-				p.onPlacementOrder(resp)
-			}
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-p.closedCh:
+			return nil
+		case <-p.clock.Tick(placementReconnectInterval):
 		}
-	}()
+	}
+}
+
+func (p *ActorPlacement) manageConnectionLoop(ctx context.Context) error {
+	if p.shutdown.Load() {
+		return nil
+	}
+
+	defer p.wg.Wait()
 
-	// Send the current host status to placement to register the member and
-	// maintain the status of member by placement.
-	p.shutdownConnLoop.Add(1)
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	serverAddr := p.serverAddr[p.serverIndex.Load()]
+	log.Infof("trying to connect to placement service: %s", serverAddr)
+
+	client, err := newPlacementClient(ctx, serverAddr, p.grpcOpts)
+	if errors.Is(err, errEstablishingTLSConn) || errors.Is(err, context.Canceled) {
+		return nil
+	}
+	if err != nil {
+		p.serverIndex.Store((p.serverIndex.Load() + 1) % int32(len(p.serverAddr)))
+		return err
+	}
+
+	log.Infof("connected to placement service at address %s", serverAddr)
```

Comment:
Nit (logs start uppercased usually):

```suggestion
	log.Info("Connected to placement service at address " + serverAddr)
```